TickSpec/LineParser.fs

```diff
@@ -39,49 +44,54 @@ let tryRegex input pattern =
 let startsWith (pattern:string) (s:string) =
     s.StartsWith(pattern, System.StringComparison.InvariantCultureIgnoreCase)
 
-let (|Scenario|_|) (s:string) =
-    let s = s.Trim()
-    if s |> startsWith "Scenario" || s |> startsWith "Story" then
-        Scenario s |> Some else None
-let (|IsBackground|_|) s =
+let (|FeatureLine|_|) s =
+    tryRegex s "^\s*Feature:\s(.*)"
+    |> Option.map (fun t -> FeatureLine (t.Trim()))
+let (|ScenarioLine|_|) (s:string) =
+    let trimmed = s.Trim()
+    let matches =
+        [ "Scenario"; "Story" ]
+        |> Seq.choose (fun x ->
+            if trimmed |> startsWith x then Some trimmed
+            else None)
+    if matches |> Seq.isEmpty then None
+    else ScenarioLine (matches |> Seq.head) |> Some
+let (|BackgroundLine|_|) s =
     tryRegex s "^\s*Background(.*)"
-    |> Option.map (fun t -> IsBackground)
+    |> Option.map (fun t -> BackgroundLine)
 let (|GivenLine|_|) s =
-    tryRegex s "^\s*Given\s+(.*)"
-    |> Option.map (fun t -> GivenLine t)
+    tryRegex s "^\s*Given\s(.*)"
+    |> Option.map (fun t -> GivenLine (t.Trim()))
 let (|WhenLine|_|) s =
-    tryRegex s "^\s*When\s+(.*)"
-    |> Option.map (fun t -> WhenLine t)
+    tryRegex s "^\s*When\s(.*)"
+    |> Option.map (fun t -> WhenLine (t.Trim()))
 let (|ThenLine|_|) s =
-    tryRegex s "^\s*Then\s+(.*)"
-    |> Option.map (fun t -> ThenLine t)
+    tryRegex s "^\s*Then\s(.*)"
+    |> Option.map (fun t -> ThenLine (t.Trim()))
 let (|AndLine|_|) s =
-    tryRegex s "^\s*And\s+(.*)"
-    |> Option.map (fun t -> AndLine t)
+    tryRegex s "^\s*And\s(.*)"
+    |> Option.map (fun t -> AndLine (t.Trim()))
 let (|ButLine|_|) s =
-    tryRegex s "^\s*But\s+(.*)"
-    |> Option.map (fun t -> ButLine t)
-let (|Row|_|) (s:string) =
+    tryRegex s "^\s*But\s(.*)"
+    |> Option.map (fun t -> ButLine (t.Trim()))
+let (|TableRowLine|_|) (s:string) =
     if s.Trim().StartsWith("|") then
         let options = System.StringSplitOptions.RemoveEmptyEntries
         let cols = s.Trim().Split([|'|'|],options)
-        let cols = cols |> Array.map (fun s -> s.Trim())
-        Row cols |> Some
+        let cols = cols |> Array.toList |> List.map (fun s -> s.Trim())
```

Comment:
`Seq.ofArray >> Seq.map >> List.ofSeq`
or 
`[for s in cols -> s.Trim()]`
or (using actie pattern above)
`[for (Trim s) in cols -> s]`